saul-core/doc/MODELS.md

```diff
@@ -1,6 +1,101 @@
-* Designing flexible learning models including various configurations such as:
 
-   * Local models i.e. single classifiers. (Learning only models (LO)).
-   * Constrained conditional models (CCM)[1] for training independent classifiers and using them jointly for global decision making in prediction time. (Learning+Inference (L+I)).
-   * Global models for joint training and joint inference (Inference-Based-Training (IBT)).
-   * Pipeline models for complex problems where the output of each layer is used as the input of the next layer.
+//Documented by Parisa Kordjamshidi
+
+Designing flexible learning models including various configurations such as:
+
+   * [Local models](#local) i.e. single classifiers. (Learning only models (LO)).
+   * [Constrained conditional models (CCM)](#L+I)[1] for training independent classifiers and using them jointly for global decision making in prediction time. (Learning+Inference (L+I)).
+   * [Global models](#IBT) for joint training and joint inference (Inference-Based-Training (IBT)).
+   * [Pipeline models](#pipeline) for complex problems where the output of each layer is used as the input of the next layer.
+
+The above mentioned paradigms can be tested using this simple badge classifier example, [here](saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/Badge/BagesApp.scala).
+<a name="local">
+##Local models.
+These models are classic classifiers. They are defined with the `Learnable` construct. This construct requires specifying
+ the name of a single output variable, that is, a label which is itself a property in the data model, and the features which is also a
+ comma separated list of properties.
+
+ ```scala
+ object ClassifierName extends Learnable (node) {
+   def label = property1
+   def feature = using(property2,property3,...)
+   //a comma separated list of properties
+ }
+ ```
+
+ For the details about the `Learnable` construct see [here](SAULLANGUAGE.md).
+
+<a name="L+I">
+##Learning+Inference models.
+These models are useful for when we need to consider the global relations between the outputs of a bunch of classifiers during the
+prediction time. Each classifier is defined with the same `Learnable` construct as a local model. In addition to the Learnable definitions, the programmer
+has the possibility to define a number of logical constraints between the output of the classifiers.
+Having the constraint definitions in place (see [here](SAULLANGUAGE.md) for syntax details), the programmer is able to define
+new constraint classifiers that use the Learnables and constraints.
```

Comment:
Either "constraint classifier" or "constrained classifier", we should always use one. I suggest the latter, as it's been used more in the repository.